File: problema6.py
nombre=input("Ingresa tu nombre: ")
genero=input("Ingresa tu género (M/F): ")

if genero.upper()=="M":
    grupo="A"
elif genero.upper()=="F":
    grupo="B"
else:
    grupo="C"

print(f"{nombre}, perteneces al grupo {grupo}.")

File: problema6.hs
import Data.Char (toLower)

asignarGrupo :: String -> String -> String
asignarGrupo nombre genero
  | map toLower genero=="mujer"="Eres del grupo A, "++nombre
  | otherwise= "Eres del grupo B, "++nombre

main :: IO ()
main = do
  putStrLn "Introduce tu nombre:"
  nombre<-getLine
  putStrLn "Introduce tu género (hombre/mujer):"
  genero<-getLine
  putStrLn (asignarGrupo nombre genero)